src/overlays/actors/ovl_Shot_Sun/z_shot_sun.c

```diff
@@ -26,17 +33,157 @@ const ActorInit Shot_Sun_InitVars = {
+    }
+
+    Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_ELF, this->actor.initPosRot.pos.x,
+                this->actor.initPosRot.pos.y, this->actor.initPosRot.pos.z, 0, 0, 0, fairyParams);
+
+    Actor_Kill(&this->actor);
+}
+
+void ShotSun_TriggerFairy(ShotSun* this, GlobalContext* globalCtx) {
+    if ((func_8005B198() == this->actor.type) || (this->timer != 0)) {
+        this->actionFunc = ShotSun_SpawnFairy;
+        this->timer = 50;
+
+        Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_DEMO_KANKYO, this->actor.initPosRot.pos.x,
+                    this->actor.initPosRot.pos.y, this->actor.initPosRot.pos.z, 0, 0, 0, 0x11);
+
+        func_80078914(&this->actor.projectedPos, NA_SE_EV_TRE_BOX_APPEAR);
+    }
+}
+
+void func_80BADF0C(ShotSun* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    s32 pad;
+    s32 params = this->actor.params & 0xFF;
+
+    if (func_800CB650(&this->actor.posRot.pos, &player->actor.posRot.pos) > 22500.0f) {
+        this->unk_1A4[0] = 0;
+    } else {
+        if (this->unk_1A4[0] == 0) {
+            if (!(player->stateFlags2 & 0x1000000)) {
+                player->stateFlags2 |= 0x800000;
+                return;
+            } else {
+                this->unk_1A4[0] = 1;
+            }
+        }
+        if (this->unk_1A4[0] == 1) {
+            func_8010BD58(globalCtx, 1U);
+            this->unk_1A4[0] = 2;
+            return;
+        }
+
+        if (this->unk_1A4[0] == 2 && globalCtx->msgCtx.unk_E3EE == 4) {
+            if ((params == 0x40 && globalCtx->msgCtx.unk_E3EC == 9) ||
+                (params == 0x41 && globalCtx->msgCtx.unk_E3EC == 0xB)) {
+                this->actionFunc = ShotSun_TriggerFairy;
+                func_80080480(globalCtx, this);
+                this->timer = 0;
+            } else {
+                this->unk_1A4[0] = 0;
+            }
+            this->unk_1A4[0] = 0;
+        }
+    }
+}
+
+void ShotSun_UpdateHyliaSun(ShotSun* this, GlobalContext* globalCtx) {
+    Vec3s cylinderPos;
+    Player* player = PLAYER;
+    EnItem00* collectible;
+    s32 pad;
+    Vec3f spawnPos;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Shot_Sun/func_80BADDCC.s")
+    if (this->collider.base.acFlags & 2) {
+        func_80078884(NA_SE_SY_CORRECT_CHIME);
+        osSyncPrintf(VT_FGCOL(CYAN) "SHOT_SUN HIT!!!!!!!\n" VT_RST);
+        if (INV_CONTENT(SLOT_ARROW_FIRE) == ITEM_NONE) {
+            Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_ITEM_ETCETERA, 700.0f, -800.0f, 7261.0f, 0, 0, 0, 7);
+            globalCtx->csCtx.segment = SEGMENTED_TO_VIRTUAL(&D_02007020);
+            if (1) {}
+            gSaveContext.cutsceneTrigger = 1;
+        } else {
+            spawnPos.x = 700.0f;
+            spawnPos.y = -800.0f;
+            spawnPos.z = 7261.0f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Shot_Sun/func_80BADE74.s")
+            collectible = Item_DropCollectible(globalCtx, &spawnPos, 0xE);
```

Comment:
the 0xE is ITEM00_MAGIC_LARGE from the item00 params enum that you moved to z64_actor